src/main/java/com/google/sps/app/servlets/NewEventServlet.java

```diff
@@ -0,0 +1,92 @@
+
+package com.google.sps.app.servlets;
+
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.OrderBy;
+import com.google.gson.Gson;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.util.Timestamps;
+import com.google.sps.model.Category;
+import com.google.sps.model.Event;
+import com.google.sps.util.UUIDs;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.commons.text.StringEscapeUtils;
+
+@WebServlet("/NewEvent")
+public class NewEventServlet extends HttpServlet {
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        String error = validateInput(request);
+        if (error.isEmpty()) {
+            // Get the value entered in the form.
+            String textValuetitle = StringEscapeUtils.escapeHtml4(request.getParameter("text-input-title"));
+            float estimatedCost = Float
+                    .parseFloat(request.getParameter("text-input-estimatedCost"));
+
+            UUID eventID = UUIDs.generateID();
+            String location = StringEscapeUtils.escapeHtml4(request.getParameter("text-input-location"));
+            String date = request.getParameter("text-input-date");
+            writeToDatastore(eventID, textValuetitle, estimatedCost, location, date);
```

Comment:
For readabilty, create a method called getEvent(request), that will extract the title, cost, location, id & return an Event object. You can then pass that object into the writeToDatsStore.

You code will be:
String error = validateInput(request);
if (error.isEmpty()) {
   Event event = getEvent(request);
   writeToDatastore(event);
   // return json
} else {
  return error;
}